desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -418,11 +416,27 @@ public void setDatastore(final Datastore datastore, boolean expandTree) {
             changePanel(AnalysisWindowPanelType.WELCOME);
         } else {
             changePanel(AnalysisWindowPanelType.EDITING_CONTEXT);
+            addTableToSource(con);
         }
 
         updateStatusLabel();
     }
 
+    private void addTableToSource(final DatastoreConnection con) {
+        if (con != null) {
+            final Schema defaultSchema = con.getSchemaNavigator().getDefaultSchema();
+            final int datastoreSize = defaultSchema.getTables().length;
+            if (datastoreSize <= 1) {
```

Comment:
I would  call this "tableCount" and do also make a equals to 1, since you might encounter schemas without tables.